diff --git a/rootdir/fstab.hi3635 b/rootdir/fstab.hi3635
index b002764..56b70a6 100755
--- a/rootdir/fstab.hi3635
+++ b/rootdir/fstab.hi3635
@@ -20,7 +20,7 @@
 /dev/block/platform/hi_mci.0/by-name/boot      		/boot          		emmc    defaults                                                 						defaults
 /dev/block/platform/hi_mci.0/by-name/recovery  		/recovery      		emmc    defaults                                                 						defaults
 
-/devices/hi_mci.1/mmc_host/mmc1/*              		auto           		vfat    defaults												voldmanaged=sdcard:auto,noemulatedsd,encryptable=userdata
+/devices/hi_mci.1/mmc_host/mmc1/*              		auto           		vfat    defaults												voldmanaged=sdcard:auto,encryptable=userdata
 /devices/hisi-usb-otg/usb1/*		      		auto           		auto    defaults												voldmanaged=usbotg:auto
 
 /dev/block/zram0             				none         		swap    defaults												zramsize=268435456
