diff --git a/sepolicy/file_contexts b/sepolicy/file_contexts
index 83ec164..720a028 100644
--- a/sepolicy/file_contexts
+++ b/sepolicy/file_contexts
@@ -1,4 +1,7 @@
 # File contexts
+/dev/imgsysbrg_vdec[0-9]*				u:object_r:video_device:s0
+/dev/jpu						u:object_r:video_device:s0
+/dev/mali0						u:object_r:gpu_device:s0
 /system/bin/glgps4752					u:object_r:gpsd_exec:s0
 /system/bin/gpsdaemon					u:object_r:gpsdaemon_exec:s0
 /vendor/bin/mac_addr_normalization			u:object_r:mac_addr_normalization_exec:s0
