diff --git a/configs/props/system.prop b/configs/props/system.prop
index a67687d..4a53759 100644
--- a/configs/props/system.prop
+++ b/configs/props/system.prop
@@ -19,6 +19,10 @@ dalvik.vm.heapmaxfree=8m
 ro.sf.lcd_density=320
 ro.opengles.version=196608
 ro.config.mmu_en=1
+ro.hardware.egl=mali
+ro.hardware.gralloc=hi3635
+ro.hardware.copybit=hi3635
+ro.hardware.memtrack=hi3635
 
 # HWUI
 ro.hwui.texture_cache_size=72
