diff --git a/mkbootimg.py b/mkbootimg.py
index c5ff1e786c5ad87313935c92746d1e1b8c04fded..0daa619e404be4966cd1b76a2f2adaae3402ff13 100644
--- a/mkbootimg.py
+++ b/mkbootimg.py
@@ -115,7 +115,10 @@ def write_header(args):
 
     args.output.write(pack('8s', BOOT_MAGIC))
     final_ramdisk_offset = (args.base + args.ramdisk_offset) if filesize(args.ramdisk) > 0 else 0
-    final_second_offset = (args.base + args.second_offset) if filesize(args.second) > 0 else 0
+    final_second_offset = (
+        args.base + args.second_offset
+        if filesize(args.second) > 0 or args.legacy_nonzero_second_addr
+        else 0)
     args.output.write(pack(
         '10I',
         filesize(args.kernel),                          # size in bytes
@@ -249,6 +252,9 @@ def parse_cmdline():
                         default=0x01000000)
     parser.add_argument('--second_offset', help='2nd bootloader offset', type=parse_int,
                         default=0x00f00000)
+    parser.add_argument('--legacy_nonzero_second_addr',
+                        help='preserve the 2nd bootloader address when its size is zero',
+                        action='store_true')
     parser.add_argument('--dtb_offset', help='dtb offset', type=parse_int, default=0x01f00000)
 
     parser.add_argument('--os_version', help='operating system version', type=parse_os_version,
