diff --git a/audio/core/all-versions/default/include/core/all-versions/default/Device.impl.h b/audio/core/all-versions/default/include/core/all-versions/default/Device.impl.h
index 230b8de..5e1c2a9 100644
--- a/audio/core/all-versions/default/include/core/all-versions/default/Device.impl.h
+++ b/audio/core/all-versions/default/include/core/all-versions/default/Device.impl.h
@@ -66,11 +66,12 @@ Return<Result> Device::initCheck() {
 }
 
 Return<Result> Device::setMasterVolume(float volume) {
-    if (mDevice->set_master_volume == NULL) {
+    if (mDevice == NULL || mDevice->set_master_volume == NULL) {
+        ALOGW("setMasterVolume not supported on this device");
         return Result::NOT_SUPPORTED;
     }
     if (!isGainNormalized(volume)) {
-        ALOGW("Can not set a master volume (%f) outside [0,1]", volume);
+        ALOGW("Invalid volume level (%f), must be in range [0,1]", volume);
         return Result::INVALID_ARGUMENTS;
     }
     return analyzeStatus("set_master_volume", mDevice->set_master_volume(mDevice, volume));
diff --git a/audio/core/all-versions/default/include/core/all-versions/default/PrimaryDevice.impl.h b/audio/core/all-versions/default/include/core/all-versions/default/PrimaryDevice.impl.h
index f269dd4..44962c8 100644
--- a/audio/core/all-versions/default/include/core/all-versions/default/PrimaryDevice.impl.h
+++ b/audio/core/all-versions/default/include/core/all-versions/default/PrimaryDevice.impl.h
@@ -36,6 +36,10 @@ Return<Result> PrimaryDevice::initCheck() {
 }
 
 Return<Result> PrimaryDevice::setMasterVolume(float volume) {
+    if (mDevice == nullptr) {
+        ALOGW("setMasterVolume called, but mDevice is NULL");
+        return Result::NOT_SUPPORTED;
+    }
     return mDevice->setMasterVolume(volume);
 }
 
