diff --git a/mozart-vendor.mk b/mozart-vendor.mk
index b33df61eb8790ab2eb753fd8c3d2d5d0ee795da1..f4de3278207d18f3b4393ddd1d6d9a67c05f28bd 100644
--- a/mozart-vendor.mk
+++ b/mozart-vendor.mk
@@ -11,9 +11,27 @@ PRODUCT_COPY_FILES += \
     $(VENDOR_PATH)/vendor/lib64/libbt-vendor.so:$(TARGET_COPY_OUT_SYSTEM)/vendor/lib64/libbt-vendor.so \
     $(VENDOR_PATH)/vendor/lib64/hw/sensorcaps.hi3635.so:$(TARGET_COPY_OUT_SYSTEM)/vendor/lib64/hw/sensorcaps.hi3635.so \
     $(VENDOR_PATH)/vendor/lib64/egl/libGLES_mali.so:$(TARGET_COPY_OUT_SYSTEM)/vendor/lib64/egl/libGLES_mali.so \
+    $(VENDOR_PATH)/lib64/hw/copybit.hi3635.so:$(TARGET_COPY_OUT_SYSTEM)/vendor/lib64/hw/copybit.hi3635.so \
+    $(VENDOR_PATH)/lib64/hw/gralloc.hi3635.so:$(TARGET_COPY_OUT_SYSTEM)/vendor/lib64/hw/gralloc.hi3635.so \
+    $(VENDOR_PATH)/lib64/hw/memtrack.hi3635.so:$(TARGET_COPY_OUT_SYSTEM)/vendor/lib64/hw/memtrack.hi3635.so \
+    $(VENDOR_PATH)/lib64/libion_mozart.so:$(TARGET_COPY_OUT_SYSTEM)/vendor/lib64/libion_mozart.so \
+    $(VENDOR_PATH)/lib64/libomx_vxd.so:$(TARGET_COPY_OUT_SYSTEM)/vendor/lib64/libomx_vxd.so \
+    $(VENDOR_PATH)/lib64/libomx_vxe.so:$(TARGET_COPY_OUT_SYSTEM)/vendor/lib64/libomx_vxe.so \
+    $(VENDOR_PATH)/lib64/libstagefrighthw.so:$(TARGET_COPY_OUT_SYSTEM)/vendor/lib64/libstagefrighthw.so \
+    $(VENDOR_PATH)/lib64/libstlport.so:$(TARGET_COPY_OUT_SYSTEM)/vendor/lib64/libstlport.so \
+    $(VENDOR_PATH)/lib64/libvpp.so:$(TARGET_COPY_OUT_SYSTEM)/vendor/lib64/libvpp.so \
     $(VENDOR_PATH)/vendor/lib64/sensorhub.hi3635.so:$(TARGET_COPY_OUT_SYSTEM)/vendor/lib64/sensorhub.hi3635.so \
     $(VENDOR_PATH)/vendor/lib/hw/sensorcaps.hi3635.so:$(TARGET_COPY_OUT_SYSTEM)/vendor/lib/hw/sensorcaps.hi3635.so \
     $(VENDOR_PATH)/vendor/lib/egl/libGLES_mali.so:$(TARGET_COPY_OUT_SYSTEM)/vendor/lib/egl/libGLES_mali.so \
+    $(VENDOR_PATH)/lib/hw/copybit.hi3635.so:$(TARGET_COPY_OUT_SYSTEM)/vendor/lib/hw/copybit.hi3635.so \
+    $(VENDOR_PATH)/lib/hw/gralloc.hi3635.so:$(TARGET_COPY_OUT_SYSTEM)/vendor/lib/hw/gralloc.hi3635.so \
+    $(VENDOR_PATH)/lib/hw/memtrack.hi3635.so:$(TARGET_COPY_OUT_SYSTEM)/vendor/lib/hw/memtrack.hi3635.so \
+    $(VENDOR_PATH)/lib/libion_mozart.so:$(TARGET_COPY_OUT_SYSTEM)/vendor/lib/libion_mozart.so \
+    $(VENDOR_PATH)/lib/libomx_vxd.so:$(TARGET_COPY_OUT_SYSTEM)/vendor/lib/libomx_vxd.so \
+    $(VENDOR_PATH)/lib/libomx_vxe.so:$(TARGET_COPY_OUT_SYSTEM)/vendor/lib/libomx_vxe.so \
+    $(VENDOR_PATH)/lib/libstagefrighthw.so:$(TARGET_COPY_OUT_SYSTEM)/vendor/lib/libstagefrighthw.so \
+    $(VENDOR_PATH)/lib/libstlport.so:$(TARGET_COPY_OUT_SYSTEM)/vendor/lib/libstlport.so \
+    $(VENDOR_PATH)/lib/libvpp.so:$(TARGET_COPY_OUT_SYSTEM)/vendor/lib/libvpp.so \
     $(VENDOR_PATH)/vendor/firmware/nvram4339_hw.txt:$(TARGET_COPY_OUT_SYSTEM)/vendor/firmware/nvram4339_hw.txt \
     $(VENDOR_PATH)/vendor/firmware/fw_bcm4339_hw.bin:$(TARGET_COPY_OUT_SYSTEM)/vendor/firmware/fw_bcm4339_hw.bin \
     $(VENDOR_PATH)/vendor/firmware/dhd_preinit_config.txt:$(TARGET_COPY_OUT_SYSTEM)/vendor/firmware/dhd_preinit_config.txt \
@@ -24,7 +42,7 @@ PRODUCT_COPY_FILES += \
     $(VENDOR_PATH)/vendor/firmware/nvram4339_hw_mozart_l.txt:$(TARGET_COPY_OUT_SYSTEM)/vendor/firmware/nvram4339_hw_mozart_l.txt \
     $(VENDOR_PATH)/vendor/firmware/fw_bcm4339_apsta_hw.bin:$(TARGET_COPY_OUT_SYSTEM)/vendor/firmware/fw_bcm4339_apsta_hw.bin \
     $(VENDOR_PATH)/vendor/firmware/BCM4334B0_002.001.013.1837.2355.hcd:$(TARGET_COPY_OUT_SYSTEM)/vendor/firmware/BCM4334B0_002.001.013.1837.2355.hcd \
-    $(VENDOR_PATH)/vendor/bin/mac_addr_normalization:$(TARGET_COPY_OUT_SYSTEM)/vendor/bin/mac_addr_normalization \
+    $(VENDOR_PATH)/vendor/bin/mac_addr_normalization:$(TARGET_COPY_OUT_SYSTEM)/vendor/bin/mac_addr_normalization_vendor \
     $(VENDOR_PATH)/lib64/liboeminfo.so:$(TARGET_COPY_OUT_SYSTEM)/lib64/liboeminfo.so \
     $(VENDOR_PATH)/lib64/libtfa9895.so:$(TARGET_COPY_OUT_SYSTEM)/lib64/libtfa9895.so \
     $(VENDOR_PATH)/lib64/lib_nve_operation.so:$(TARGET_COPY_OUT_SYSTEM)/lib64/lib_nve_operation.so \
@@ -33,6 +51,7 @@ PRODUCT_COPY_FILES += \
     $(VENDOR_PATH)/lib64/libaudio_custom.so:$(TARGET_COPY_OUT_SYSTEM)/lib64/libaudio_custom.so \
     $(VENDOR_PATH)/lib64/libaes.so:$(TARGET_COPY_OUT_SYSTEM)/lib64/libaes.so \
     $(VENDOR_PATH)/lib64/hw/flp.hi3635.so:$(TARGET_COPY_OUT_SYSTEM)/lib64/hw/flp.hi3635.so \
+    $(VENDOR_PATH)/lib64/hw/copybit.hi3635.so:$(TARGET_COPY_OUT_SYSTEM)/lib64/hw/copybit.hi3635.so \
     $(VENDOR_PATH)/lib64/hw/memtrack.hi3635.so:$(TARGET_COPY_OUT_SYSTEM)/lib64/hw/memtrack.hi3635.so \
     $(VENDOR_PATH)/lib64/hw/gralloc.hi3635.so:$(TARGET_COPY_OUT_SYSTEM)/lib64/hw/gralloc.hi3635.so \
     $(VENDOR_PATH)/lib64/hw/audio.primary.hi3635.so:$(TARGET_COPY_OUT_SYSTEM)/lib64/hw/audio.primary.hi3635.so \
@@ -43,7 +62,7 @@ PRODUCT_COPY_FILES += \
     $(VENDOR_PATH)/lib64/libaudioroute.so:$(TARGET_COPY_OUT_SYSTEM)/lib64/libaudioroute.so \
     $(VENDOR_PATH)/lib64/libstagefrighthw.so:$(TARGET_COPY_OUT_SYSTEM)/lib64/libstagefrighthw.so \
     $(VENDOR_PATH)/lib64/libstlport.so:$(TARGET_COPY_OUT_SYSTEM)/lib64/libstlport.so \
-    $(VENDOR_PATH)/lib64/libion.so:$(TARGET_COPY_OUT_SYSTEM)/lib64/libion.so \
+    $(VENDOR_PATH)/lib64/libion_mozart.so:$(TARGET_COPY_OUT_SYSTEM)/lib64/libion_mozart.so \
     $(VENDOR_PATH)/lib64/libvpp.so:$(TARGET_COPY_OUT_SYSTEM)/lib64/libvpp.so \
     $(VENDOR_PATH)/lib64/libomx_vxd.so:$(TARGET_COPY_OUT_SYSTEM)/lib64/libomx_vxd.so \
     $(VENDOR_PATH)/lib64/libmax98925.so:$(TARGET_COPY_OUT_SYSTEM)/lib64/libmax98925.so \
@@ -57,6 +76,7 @@ PRODUCT_COPY_FILES += \
     $(VENDOR_PATH)/lib/libbalong_audio_ril.so:$(TARGET_COPY_OUT_SYSTEM)/lib/libbalong_audio_ril.so \
     $(VENDOR_PATH)/lib/libaudio_custom.so:$(TARGET_COPY_OUT_SYSTEM)/lib/libaudio_custom.so \
     $(VENDOR_PATH)/lib/libaes.so:$(TARGET_COPY_OUT_SYSTEM)/lib/libaes.so \
+    $(VENDOR_PATH)/lib/hw/copybit.hi3635.so:$(TARGET_COPY_OUT_SYSTEM)/lib/hw/copybit.hi3635.so \
     $(VENDOR_PATH)/lib/hw/memtrack.hi3635.so:$(TARGET_COPY_OUT_SYSTEM)/lib/hw/memtrack.hi3635.so \
     $(VENDOR_PATH)/lib/hw/gralloc.hi3635.so:$(TARGET_COPY_OUT_SYSTEM)/lib/hw/gralloc.hi3635.so \
     $(VENDOR_PATH)/lib/hw/audio.primary.hi3635.so:$(TARGET_COPY_OUT_SYSTEM)/lib/hw/audio.primary.hi3635.so \
@@ -68,7 +88,7 @@ PRODUCT_COPY_FILES += \
     $(VENDOR_PATH)/lib/libaudioroute.so:$(TARGET_COPY_OUT_SYSTEM)/lib/libaudioroute.so \
     $(VENDOR_PATH)/lib/libstagefrighthw.so:$(TARGET_COPY_OUT_SYSTEM)/lib/libstagefrighthw.so \
     $(VENDOR_PATH)/lib/libstlport.so:$(TARGET_COPY_OUT_SYSTEM)/lib/libstlport.so \
-    $(VENDOR_PATH)/lib/libion.so:$(TARGET_COPY_OUT_SYSTEM)/lib/libion.so \
+    $(VENDOR_PATH)/lib/libion_mozart.so:$(TARGET_COPY_OUT_SYSTEM)/lib/libion_mozart.so \
     $(VENDOR_PATH)/lib/libvpp.so:$(TARGET_COPY_OUT_SYSTEM)/lib/libvpp.so \
     $(VENDOR_PATH)/lib/libomx_vxd.so:$(TARGET_COPY_OUT_SYSTEM)/lib/libomx_vxd.so \
     $(VENDOR_PATH)/lib/libhuaweiaudioalgoservice.so:$(TARGET_COPY_OUT_SYSTEM)/lib/libhuaweiaudioalgoservice.so \
