diff --git a/mozart-vendor.mk b/mozart-vendor.mk
index b33df61..21d5e73 100644
--- a/mozart-vendor.mk
+++ b/mozart-vendor.mk
@@ -11,9 +11,27 @@ PRODUCT_COPY_FILES += \
     $(VENDOR_PATH)/vendor/lib64/libbt-vendor.so:$(TARGET_COPY_OUT_SYSTEM)/vendor/lib64/libbt-vendor.so \
     $(VENDOR_PATH)/vendor/lib64/hw/sensorcaps.hi3635.so:$(TARGET_COPY_OUT_SYSTEM)/vendor/lib64/hw/sensorcaps.hi3635.so \
     $(VENDOR_PATH)/vendor/lib64/egl/libGLES_mali.so:$(TARGET_COPY_OUT_SYSTEM)/vendor/lib64/egl/libGLES_mali.so \
+    $(VENDOR_PATH)/lib64/hw/copybit.hi3635.so:$(TARGET_COPY_OUT_SYSTEM)/vendor/lib64/hw/copybit.hi3635.so \
+    $(VENDOR_PATH)/lib64/hw/gralloc.hi3635.so:$(TARGET_COPY_OUT_SYSTEM)/vendor/lib64/hw/gralloc.hi3635.so \
+    $(VENDOR_PATH)/lib64/hw/memtrack.hi3635.so:$(TARGET_COPY_OUT_SYSTEM)/vendor/lib64/hw/memtrack.hi3635.so \
+    $(VENDOR_PATH)/lib64/libion.so:$(TARGET_COPY_OUT_SYSTEM)/vendor/lib64/libion.so \
+    $(VENDOR_PATH)/lib64/libomx_vxd.so:$(TARGET_COPY_OUT_SYSTEM)/vendor/lib64/libomx_vxd.so \
+    $(VENDOR_PATH)/lib64/libomx_vxe.so:$(TARGET_COPY_OUT_SYSTEM)/vendor/lib64/libomx_vxe.so \
+    $(VENDOR_PATH)/lib64/libstagefrighthw.so:$(TARGET_COPY_OUT_SYSTEM)/vendor/lib64/libstagefrighthw.so \
+    $(VENDOR_PATH)/lib64/libstlport.so:$(TARGET_COPY_OUT_SYSTEM)/vendor/lib64/libstlport.so \
+    $(VENDOR_PATH)/lib64/libvpp.so:$(TARGET_COPY_OUT_SYSTEM)/vendor/lib64/libvpp.so \
     $(VENDOR_PATH)/vendor/lib64/sensorhub.hi3635.so:$(TARGET_COPY_OUT_SYSTEM)/vendor/lib64/sensorhub.hi3635.so \
     $(VENDOR_PATH)/vendor/lib/hw/sensorcaps.hi3635.so:$(TARGET_COPY_OUT_SYSTEM)/vendor/lib/hw/sensorcaps.hi3635.so \
     $(VENDOR_PATH)/vendor/lib/egl/libGLES_mali.so:$(TARGET_COPY_OUT_SYSTEM)/vendor/lib/egl/libGLES_mali.so \
+    $(VENDOR_PATH)/lib/hw/copybit.hi3635.so:$(TARGET_COPY_OUT_SYSTEM)/vendor/lib/hw/copybit.hi3635.so \
+    $(VENDOR_PATH)/lib/hw/gralloc.hi3635.so:$(TARGET_COPY_OUT_SYSTEM)/vendor/lib/hw/gralloc.hi3635.so \
+    $(VENDOR_PATH)/lib/hw/memtrack.hi3635.so:$(TARGET_COPY_OUT_SYSTEM)/vendor/lib/hw/memtrack.hi3635.so \
+    $(VENDOR_PATH)/lib/libion.so:$(TARGET_COPY_OUT_SYSTEM)/vendor/lib/libion.so \
+    $(VENDOR_PATH)/lib/libomx_vxd.so:$(TARGET_COPY_OUT_SYSTEM)/vendor/lib/libomx_vxd.so \
+    $(VENDOR_PATH)/lib/libomx_vxe.so:$(TARGET_COPY_OUT_SYSTEM)/vendor/lib/libomx_vxe.so \
+    $(VENDOR_PATH)/lib/libstagefrighthw.so:$(TARGET_COPY_OUT_SYSTEM)/vendor/lib/libstagefrighthw.so \
+    $(VENDOR_PATH)/lib/libstlport.so:$(TARGET_COPY_OUT_SYSTEM)/vendor/lib/libstlport.so \
+    $(VENDOR_PATH)/lib/libvpp.so:$(TARGET_COPY_OUT_SYSTEM)/vendor/lib/libvpp.so \
     $(VENDOR_PATH)/vendor/firmware/nvram4339_hw.txt:$(TARGET_COPY_OUT_SYSTEM)/vendor/firmware/nvram4339_hw.txt \
     $(VENDOR_PATH)/vendor/firmware/fw_bcm4339_hw.bin:$(TARGET_COPY_OUT_SYSTEM)/vendor/firmware/fw_bcm4339_hw.bin \
     $(VENDOR_PATH)/vendor/firmware/dhd_preinit_config.txt:$(TARGET_COPY_OUT_SYSTEM)/vendor/firmware/dhd_preinit_config.txt \
@@ -33,6 +51,7 @@ PRODUCT_COPY_FILES += \
     $(VENDOR_PATH)/lib64/libaudio_custom.so:$(TARGET_COPY_OUT_SYSTEM)/lib64/libaudio_custom.so \
     $(VENDOR_PATH)/lib64/libaes.so:$(TARGET_COPY_OUT_SYSTEM)/lib64/libaes.so \
     $(VENDOR_PATH)/lib64/hw/flp.hi3635.so:$(TARGET_COPY_OUT_SYSTEM)/lib64/hw/flp.hi3635.so \
+    $(VENDOR_PATH)/lib64/hw/copybit.hi3635.so:$(TARGET_COPY_OUT_SYSTEM)/lib64/hw/copybit.hi3635.so \
     $(VENDOR_PATH)/lib64/hw/memtrack.hi3635.so:$(TARGET_COPY_OUT_SYSTEM)/lib64/hw/memtrack.hi3635.so \
     $(VENDOR_PATH)/lib64/hw/gralloc.hi3635.so:$(TARGET_COPY_OUT_SYSTEM)/lib64/hw/gralloc.hi3635.so \
     $(VENDOR_PATH)/lib64/hw/audio.primary.hi3635.so:$(TARGET_COPY_OUT_SYSTEM)/lib64/hw/audio.primary.hi3635.so \
@@ -57,6 +76,7 @@ PRODUCT_COPY_FILES += \
     $(VENDOR_PATH)/lib/libbalong_audio_ril.so:$(TARGET_COPY_OUT_SYSTEM)/lib/libbalong_audio_ril.so \
     $(VENDOR_PATH)/lib/libaudio_custom.so:$(TARGET_COPY_OUT_SYSTEM)/lib/libaudio_custom.so \
     $(VENDOR_PATH)/lib/libaes.so:$(TARGET_COPY_OUT_SYSTEM)/lib/libaes.so \
+    $(VENDOR_PATH)/lib/hw/copybit.hi3635.so:$(TARGET_COPY_OUT_SYSTEM)/lib/hw/copybit.hi3635.so \
     $(VENDOR_PATH)/lib/hw/memtrack.hi3635.so:$(TARGET_COPY_OUT_SYSTEM)/lib/hw/memtrack.hi3635.so \
     $(VENDOR_PATH)/lib/hw/gralloc.hi3635.so:$(TARGET_COPY_OUT_SYSTEM)/lib/hw/gralloc.hi3635.so \
     $(VENDOR_PATH)/lib/hw/audio.primary.hi3635.so:$(TARGET_COPY_OUT_SYSTEM)/lib/hw/audio.primary.hi3635.so \
